src/core/storage/fileio/s3_api.cpp

```diff
@@ -148,6 +148,27 @@ S3Client init_aws_sdk_with_turi_env(s3url& parsed_url) {
   // TODO: add proxy support
   // clientConfiguration.proxyHost = proxy.c_str();
 
+  // set path or file for ssl certs
+  if (!get_alternative_ssl_cert_file().empty()) {
+    auto fstaus = get_file_status(get_alternative_ssl_cert_file());
+    ASSERT_TRUE(fstaus.second.empty());
+    if (fstaus.first == file_status::REGULAR_FILE) {
+      clientConfiguration.caFile = get_alternative_ssl_cert_file().c_str();
+    } else {
+      log_and_throw("Invalid file for alternative SSL certificate");
```

Comment:
Let's give a little more detail in the error message here. Something like:
`Invalid file for alternative SSL certificate. Value of TURI_FILEIO_ALTERNATIVE_SSL_CERT_FILE must be a regular file.`

Including the value of `get_alternative_ssl_cert_file()` in the error message would also be nice.

Reply:
right, I should give some care to customers.